src/vivarium_inputs/gbd_service.py

```
@@ -0,0 +1,69 @@
+import gzip
+from loguru import logger
+import pandas as pd
+from pathlib import Path
+import requests as req
+from typing import List, Callable, TypeVar
+from urllib.parse import urlencode
+
+
+SERVICE_VERSION = '1'
+BASE_URL = f'http://microsim-rancher-p01.hosts.ihme.washington.edu:5000/v{SERVICE_VERSION}'
+ARTIFACT_FOLDER = Path('/share/costeffectiveness/artifacts/')
+
+ENDPOINT_DRAWS = 'draws'
+ENDPOINT_SUMMARY = 'summary'
+ENDPOINT_METADATA = 'metadata'
+
+
+T = TypeVar('T')
+
+
+class GbdServiceError(Exception):
+    """Error for failures accessing GBD data."""
+    pass
+
+
+def _dataframe_from_response(resp) -> pd.DataFrame:
+    if resp.ok:
+        # return pickle.loads(gzip.decompress(resp.content))
+        return resp.content.decode("utf-8")
+        # return pd.read_csv(io.BytesIO(gzip.decompress(resp.content)), index_col=0)
+        # return pd.read_json(resp.content)
+    else:
+        logger.error(f'GbdServiceError: http response code {resp}')
```

You can just raise the exception here and the logger will dump the stack trace along with the error message.